### ISSUE TITLE
chunker overlay's reporting of transfer information is not super useful always, and chunker interact with --transfers in a funky way

### ISSUE THREAD

**Issue by left1000** [rclone/rclone#7210]
![chunkercouldbemorreliable](https://github.com/mikf/gallery-dl/assets/16432525/e074e86c-ffc6-40af-9728-5e7bddf006c1)

As you can see here the chunker overlay uploads chunkers then renames them via a server-side move which doubles the reported bandwidth and or filesize transferred and makes bandwidth reporting inaccurate. Now I could do some math to determine the bandwidth usage myself, BUT, if the filesize transfered data measurement is wrong, my math will result in a wrong average bandwidth used result also.

Also maybe even more critically... I find that the chunker overlay almost randomly multiplies the --transfers flag.

I think the correct function would be for a manually entered --transfers flag to overwrite some of chunker's choices. 
If I set --transfers 8 and then upload 8 files that are broken into 16 total ChunkerOverlayFileChunks; it will make rclone function as if I had set -transfers 16... And if I'm trying to carefully calibrate my rclone copy/sync command in order to properly fit inside some draconian api call restriction throttling.. I'd rather in the above example for rclone to treat --transfers 8 like --transfers 4. But I can't just type --transfers 4 because if I'm uploading even larger files maybe the 8 files are broken into 80 ChunkerOverlayFileChunks and even if I entered --transfers 4 it would behave as if I'd entered --transfers 40
So ideally when a chunker overlay is told to upload 8 files at once, that should be 8 ChunkerOverlayFileChunks. AKA, --transfers 8 always means to transfer 8 source files at once. It also usually means to transfers 8 destination files at once. But in a chunker overlay those two 8's are not equal. --transfers 8 in rclone needs to always transfer 8 destination files. Which is almost always 8 source files. But inside a chunkeroverlay, 8 destination files could be anywhere from 1-8 source files. Does that make any sense to anyone but me?

TLDR: the chunker overlay is harder to control, and harder to monitor than other overlays. I think if some genius reads this message, they could think of a way to improve it :)

**Comment by kapitainsky** [comment]
Transfers reporting is already fixed and will be included in v1.64. If you want you can try it now. Download fixed version from here:

https://beta.rclone.org/branch/fix-7183-server-side-accounting/v1.64.0-beta.7191.b3c480be2.fix-7183-server-side-accounting/

**Comment by left1000** [comment]
The accounting being already fixed is nice, but I'm still worried about the power of the --transfers flag being reduced. This somehow significantly impacts box.com for me at the moment.

Fixed accounting would've helped me spot the problem more easily, but now that I've already spotted the problem I don't need immediate help spotting the problem :)

**Comment by ncw** [comment]
> As you can see here the chunker overlay uploads chunkers then renames them via a server-side move which doubles the reported bandwidth and or filesize transferred and makes bandwidth reporting inaccurate.

As noted above this is being fixed in #7183

> But inside a chunkeroverlay, 8 destination files could be anywhere from 1-8 source files. Does that make any sense to anyone but me?

It makes sense to me :-)

I think chunker should be 1 transfer in to 1 transfer out as it uploads the chunks sequentially. However it may overlap blocks slightly which might be counting as 2 - not sure. It definitely shouldn't be more than 1:2 and should be 1:1 nearly all the time.

**Comment by left1000** [comment]
It doesn't make sense that chunker overlay is acting like a multiplier on --transfers, that's absolutely true, and maybe I'm wrong, but in testing, on box.com the chunker overlay effects performance as if chunker overlay were multiplying --transfers.

I think though there are some like confusing bits about it. box.com throttles API per minute according to their website, but it also says it allows for increased burst API usage. All of which is true. So the math is like impossible. Because if you are using too many api calls you're not instantly punished, which makes testing it a little harder.

**Comment by ncw** [comment]
> It doesn't make sense that chunker overlay is acting like a multiplier on --transfers, that's absolutely true, and maybe I'm wrong, but in testing, on box.com the chunker overlay effects performance as if chunker overlay were multiplying --transfers.

Chunker will definitely be using more API calls. Exactly how many more is difficult to tell and whether this is important to box is even harder to tell. But I'll definitely stand behind chunker using more API calls!
 
> I think though there are some like confusing bits about it. box.com throttles API per minute according to their website, but it also says it allows for increased burst API usage. All of which is true. So the math is like impossible. Because if you are using too many api calls you're not instantly punished, which makes testing it a little harder.

Yes indeed!

You can measure number of API calls using `-vv --dump headers` and grepping the logs. (I should probably make a stat for this since it would be easy to count). However all API calls are unlikely to be counted equally so this may be harder than it first appears.

**Comment by left1000** [comment]
Okay so, I stopped using chunker overlay (working on smaller files now.)

This is somehow the result. 
<img width="1280" alt="Untitled" src="https://github.com/rclone/rclone/assets/16432525/da3373b7-d5d6-4653-80cc-93826f3e2a3c">


In other words, the bandwidth reporting information was wrong for all the reasons above, but that wrongness is/was EASY to use my brain to comprehend. Yet. This version of rclone is somehow full of OTHER unexplained mysterious bandwidth inconsistencies.

The thing is I have no clue why, this is rclone 1.63
The command is google to box, uses --drive-chunk-size and --tpslimit and defaults otherwise.

The only thing at all unique about this install/config/command is that I'm using my own box.com api secret key in this example. Anyone have a clue why the bandwidth per file added together doesn't equal the minute's summarized speed?
